apiserver/facades/agent/credentialvalidator/credentialvalidator.go

```diff
@@ -204,19 +234,20 @@ func (api *CredentialValidatorAPI) cloudSupportsNoAuth(ctx context.Context, clou
 // WatchModelCredential returns a NotifyWatcher that watches what cloud credential a model uses.
 func (api *CredentialValidatorAPI) WatchModelCredential(ctx context.Context) (params.NotifyWatchResult, error) {
 	result := params.NotifyWatchResult{}
-	m, err := api.backend.Model()
+	modelService := api.modelService
+	watch, err := modelService.WatchModelCloudCredential(ctx, api.modelUUID)
 	if err != nil {
 		return result, apiservererrors.ServerError(err)
 	}
-	watch := m.WatchModelCredential()
 
 	// Consume the initial event. Technically, API calls to Watch
 	// 'transmit' the initial event in the Watch response. But
 	// NotifyWatchers have no state to transmit.
 	if _, ok := <-watch.Changes(); ok {
 		result.NotifyWatcherId = api.resources.Register(watch)
+		api.logger.Infof(ctx, "WatchModelCredential success for modelUUID %q", api.modelUUID)
 	} else {
-		err = watcher.EnsureErr(watch)
+		err = watch.Wait()
```

Comment:
Don't do this, use `EnsureRegisterWatcher` and the watcher registry

Reply:
Refactored with thanks ian!